crates/bevy_asset/src/http_source.rs

```diff
@@ -0,0 +1,262 @@
+use crate::io::{AssetReader, AssetReaderError, Reader};
+use crate::io::{AssetSource, PathStream};
+use crate::AssetApp;
+use alloc::boxed::Box;
+use bevy_app::{App, Plugin};
+use bevy_tasks::ConditionalSendFuture;
+use std::path::{Path, PathBuf};
+
+/// Adds the `http` and `https` asset sources to the app.
+///
+/// NOTE: Make sure to add this plugin *before* `AssetPlugin` to properly register http asset sources.
+///
+/// Any asset path that begins with `http` (when the `http` feature is enabled) or `https` (when the
+/// `https` feature is enabled) will be loaded from the web via `fetch`(wasm) or `ureq`(native).
+pub struct HttpSourcePlugin;
+
+impl Plugin for HttpSourcePlugin {
+    fn build(&self, app: &mut App) {
+        #[cfg(feature = "http")]
+        app.register_asset_source(
+            "http",
+            AssetSource::build()
+                .with_reader(|| Box::new(HttpSourceAssetReader::Http))
+                .with_processed_reader(|| Box::new(HttpSourceAssetReader::Http)),
+        );
+
+        #[cfg(feature = "https")]
+        app.register_asset_source(
+            "https",
+            AssetSource::build()
+                .with_reader(|| Box::new(HttpSourceAssetReader::Https))
+                .with_processed_reader(|| Box::new(HttpSourceAssetReader::Https)),
+        );
+    }
+}
+
+impl Default for HttpSourcePlugin {
+    fn default() -> Self {
+        Self
+    }
+}
+
+/// Asset reader that treats paths as urls to load assets from.
+pub enum HttpSourceAssetReader {
```

Comment:
Can we add a doc comment here saying that this source reader does not support directories? I don't think that's a limitation we can fix, but it's one I want documented.